### PR TITLE
[image] use scratch on final stage

### PR DESCRIPTION
**Pull request by moeryomenko**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -4,9 +4,21 @@ WORKDIR /app
 COPY main.go main.go
 RUN CGO_ENABLED=0 GOOS=linux go build -a -o ease.exec main.go
 
-FROM alpine:3.12.4
+# Create the user and group files that will be used in the running container to
+# run the process as an unprivileged user.
+RUN mkdir /user \
+    && echo '1000:x:65534:65534:1000:/:' > /user/passwd \
+    && echo '1000:x:65534:' > /user/group
```
Comment (cristaloleg) [inline Dockerfile:11]: To be honest I preferred something slightly simpler:

```
RUN adduser -S -D -u 1000 nonroot && chown -R nonroot /app

USER nonroot
```

:)

File: Dockerfile [hunk 2]
```diff
@@ -4,9 +4,21 @@ WORKDIR /app
...
+
+FROM scratch
+
+# Import the user and group files from the build stage.
+COPY --from=build_go /user/group /user/passwd /etc/
```
Comment (cristaloleg) [inline Dockerfile:16]: Regarding comment above - this will not be needed AFAIK

**Review by cristaloleg — COMMENTED** [review]
Probably it's fine but what we're trying to solve?

**Comment by moeryomenko** [comment]
> Probably it's fine but what we're trying to solve?

`Just for fun`. I think that for a container with one static binary, `scratch` environment will be enough, also it's reduce image size

**Comment by cristaloleg** [comment]
Looks like it's not so important. `alpine` works good enough and size is small enough to leave it.

**Comment by moeryomenko** [comment]
Ok, it's up to you

**Pull request closed without merging**